tools/clang/test/HLSLFileCheck/passes/hl/sroa_hlsl/memcpy_digv_replacement.hlsl

```diff
@@ -0,0 +1,30 @@
+// RUN: %dxc -T vs_6_0 -Zi %s | FileCheck %s
+
+// Test replacement of static global in a case where SROA may try to
+// replace the debug info for the flattened global variable that no longer
+// has any users
+
+// CHECK: ReplacementVar.0 = internal unnamed_addr constant [2 x float]
+// CHECK: ReplacementVar.1 = internal unnamed_addr constant [2 x float]
+// CHECK: ReplacementVar.2 = internal unnamed_addr constant [2 x float]
+// CHECK: ReplacementVar.3 = internal unnamed_addr constant [2 x float]
+
+
+// CHECK: @main
+
+
+static const float4 ReplacementVar[2] = { float4(1.0f, 2.0f, 3.0f, 4.0f),
+                                          float4(9.0f, 8.0f, 7.0f, 6.0f) };
+
+static float4 UnusedVar[2];
+
+float4 main(int ix : I) : SV_Position
+{
+    UnusedVar = ReplacementVar;
+    return  UnusedVar[ix];
+}
+
+
+// Exclude quoted source file (see readme)
+// CHECK-LABEL: {{!"[^"]*\\0A[^"]*"}}
```

Comment:
What if we check for the DIGV pointing to `ReplacementVar`?

Reply:
Are there two sets of DIGV's both pointing to `ReplacementVar.*`?

Reply:
No, the change means that the DIGV doesn't get flattened. So there is one lingering reference to the unusedvar

Reply:
I can add checks for the DIGVs

Reply:
So does this test cause the assert to fire on debug build?  That means we can't pass the tests on a debug build.
If we have an identifiable case that hits an assert, we should see what we can change to fix that, shouldn't we?